server/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -575,7 +588,19 @@ private IndexStorePlugin.DirectoryFactory getDirectoryFactory(
         final DirectoryWrapper directoryWrapper = this.indexDirectoryWrapper.get();
         assert frozen.get() : "IndexModule configuration not frozen";
         if (directoryWrapper != null) {
-            return (idxSettings, shardPath) -> directoryWrapper.apply(factory.newDirectory(idxSettings, shardPath));
+            return new IndexStorePlugin.DirectoryFactory() {
+
+                @Override
+                public Directory newDirectory(IndexSettings indexSettings, ShardPath shardPath) throws IOException {
+                    return directoryWrapper.wrap(factory.newDirectory(indexSettings, shardPath), null);
```

Comment:
My only question really is, can we do without the `null` passing override? I would assume it would only be used in tests now?

Reply:
Sure, I pushed a60cd78797daaa870cf0d22d855d141dc30db264